cleanlab/object_detection/rank.py

```diff
@@ -0,0 +1,1157 @@
+    """
+
+    assert_valid_inputs(
+        annotations=annotations, results=results, method=method, threshold=threshold
+    )
+
+    return _compute_label_quality_scores(
+        annotations=annotations,
+        results=results,
+        method=method,
+        threshold=threshold,
+        verbose=verbose,
+    )
+
+
+def issues_from_scores(label_quality_scores: np.ndarray, *, threshold: float = 0.1) -> np.ndarray:
+    """Returns a list of indices images with issues sorted from most to least severe.
+
+    Parameters
+    ----------
+    label_quality_scores:
+        Array of shape ``(N, )`` of scores between 0 and 1, one per image in the dataset.
+        Lower scores indicate images are more likely to contain a label issue.
+
+    threshold:
+        Label quality scores above the threshold are not considered as issues and their indices are omited from the return
+
+    Returns
+    ---------
+    issue_indices:
+        Array of issue indices sorted from most to least severe who's label quality scores fall below the threshold if one is provided.
+    """
+
+    if threshold > 1.0:
+        raise ValueError(
+            f"""
+            Threshold is a cutoff of label_quality_scores and therefore should be <= 1.
+            """
+        )
+
+    issue_indices = np.argwhere(label_quality_scores < threshold).flatten()
+    issue_vals = label_quality_scores[issue_indices]
+    sorted_idx = issue_vals.argsort()
+    return issue_indices[sorted_idx]
+
+
+def _compute_label_quality_scores(
+    annotations: List[Dict[Any, Any]],
+    results: List[np.ndarray],
+    *,
+    method: str = "map",
+    threshold: Optional[float] = None,
+    verbose: bool = True,
+) -> np.ndarray:
+    """Internal function to prune extra bounding boxes and compute label quality scores based on passed in method."""
+    if threshold is not None:
+        results = _prune_by_threshold(results=results, threshold=threshold, verbose=verbose)
+
+    if method == "map":
+        scores = _calculate_map(results, annotations)
+
+    return scores
+
+
+def _prune_by_threshold(
+    results: List[np.ndarray], threshold: float, verbose: bool = True
+) -> Tuple[List[Dict[Any, Any]], List[np.ndarray]]:
+    """Removes predicted bounding boxes from results who's pred_prob is below the cuttoff threshold."""
+
+    results_copy = copy.deepcopy(results)
+    num_ann_to_zero = 0
+    total_ann = 0
+    for idx_results, result in enumerate(results_copy):
+        for idx_class, class_result in enumerate(result):
+            filtered_class_result = class_result[class_result[:, -1] >= threshold]
+            if len(class_result) > 0:
+                total_ann += 1
+                if len(filtered_class_result) == 0:
+                    num_ann_to_zero += 1
+
+            results_copy[idx_results][idx_class] = filtered_class_result
+
+    p_ann_pruned = total_ann and num_ann_to_zero / total_ann or 0  # avoid division by zero
+    if p_ann_pruned > 0.97:
```

Comment:
define variable for this magic number, and add comment next to the variable saying why you chose this value